src/Formats/SchemaInferenceUtils.cpp

```diff
@@ -307,6 +318,54 @@ namespace
         type_indexes.erase(TypeIndex::UInt64);
     }
 
+    /// if setting 'try_infer_variant' is true then we convert to type variant.
+    void transformVariant(DataTypes & data_types, TypeIndexesSet & type_indexes)
+    {
+        auto typesAreEqual = checkIfTypesAreEqual(data_types);
+        auto typesContainVariant = checkIfTypesContainVariant(data_types);
+        if (typesAreEqual)
+            return;
```

Comment:
```suggestion
        auto types_are_equal = checkIfTypesAreEqual(data_types);
        auto types_contain_variant = checkIfTypesContainVariant(data_types);
        if (types_are_equal)
            return;
```
We don't use `camelCase` for variable names